lesson12.md

```diff
@@ -0,0 +1,359 @@
+<!-- .slide: id="lesson12" -->
+
+# Basic Frontend - Spring 2024
+
+Lesson 12, Tuesday, 2024-05-07
+
+---
+
+### Recap
+
+What do you remember about changing HTML from JavaScript?
+
+---
+
+<!-- .slide: id="arrays" -->
+
+# Array
+
+---
+
+### Arrays
+
+An `array` is a container type that holds multiple values:
+
+```js
+// we create an empty array using []
+let emptyArray = [];
+
+// we put the values we want in square brackets
+// separated by commas
+let ages = [19, 33, 25, 40];
+let cities = ['London', 'Paris', 'Berlin'];
+```
+
+---
+
+Array can hold any type of value:
+
+```js
+let prices = [0.99, 1.49];
+```
+
+And any quantity:
+
+```js
+// I only have one favorite food
+let favoriteFoods = ['Pizza'];
+// An array holding 26 letters of the alphabet
+let alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l']; //TODO
+```
+
+---
+
+### Analogy: Bookshelf
+
+A bookshelf can be thought of as an array of books
+
+![bookshelf](images/bookshelf.jpg)
+
+```js
+let books = ['Harry Potter', 'The Hobbit', 'Game of Thrones'];
+```
+
+---
+
+### Usage
+
+Can you think of use cases for arrays? Where would we use it?
+
+- a todo list
+<!-- .element: class="fragment" -->
+- a shopping list
+<!-- .element: class="fragment" -->
+- ingredients in a recipe
+<!-- .element: class="fragment" -->
+- the students in this class
+<!-- .element: class="fragment" -->
+- the numbers from 1 to 10
+<!-- .element: class="fragment" -->
+
+---
+
+### Accessing elements
```

Comment:
Oh, and I thought about a quiz:

```js
const myArray = [42, 42, 42]
```

How many times is 42 present?
As to show that duplicates are kept.